### DISCUSSION TITLE
Pixel 4a - Getting idle mode to work

### DISCUSSION THREAD

**Discussion by Bryantdl7** [VR-25/acc#177] [Show and tell]
Getting idle mode to reliably work on a pixel 4a is a bit of a challenge compared to easier devices like motorola or samsung. 

I will find at times something works one day, then after x amount of uptime it stops.... very annoying! 

Anywho, I think I have finally cracked it, I am running a stock config other than three options. Obviously these aren't the proper syntax to paste in, but if you know how to edit the config, this should help. 

charging switch: 7150_bms/charge_disable 
idle mode: true 
current workaround: true

For reference this is on LineageOS 19.1, pixel 4a sunfish. 

EDIT: I will keep updating this post as I keep battling it. Seems like I still haven't fully gotten it to cooperate. Any other Pixel 4a users feel free to chime in.

**Comment by Bryantdl7** [comment]
Essentially the battle I'll have is I let the phone sit, and idle mode and the charge limit both break, and are only fixed by a full OS reboot. Nothing short of an OS reboot fixes it whether rebooting acc, unplugging and replugging etc. Good times...

**Reply by Bryantdl7** [reply]
The most recent time it was after five minutes or so, and I rebooted again, and now we are good. I will continue to rant on how this is going, but I am super close, I can tell. 

Maybe I need to limit the input amperage to 1000ma....


Actually, limiting the amperage immediately broke my charge limit. I think I am onto something. The second I set the limit back to acc default, we went back to 0mA charging after an acc reboot.

**Comment by VR-25** [comment]
Hi, upgrade to the latest version and refer to #201.

**Comment by Meeeds** [comment]
Hello, 

I want to jump on this thread to clarify that current version of acc (v2025.5.18-dev (202505180))
Works perfectly with google pixel 4a (using rooted stock android) 

acc automatically used `chargingSwitch=(/sys/kernel/debug/google_charger/chg_suspend 0 1)`

And it is able to put my phone in Idle 

```

# acc -i

level 54%
status Idle
temp 35℃

current_now 0.00A
voltage_now 3.79V
power_now 0.00W

charge_type USB_DCP

dc/online 0
pc_port/online 0
sm7150_bms/online 1
tcpm-source-psy-usbpd0/online 1
usb/online 1

```
